zellij-tile/src/data.rs

```diff
@@ -67,6 +67,64 @@ impl Default for InputMode {
     }
 }
 
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum Theme {
+    Light,
+    Dark,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum PaletteSource {
+    Default,
+    Xresources,
+}
+pub mod colors {
+    pub const WHITE: (u8, u8, u8) = (238, 238, 238);
+    pub const GREEN: (u8, u8, u8) = (175, 255, 0);
+    pub const GRAY: (u8, u8, u8) = (68, 68, 68);
+    pub const BRIGHT_GRAY: (u8, u8, u8) = (138, 138, 138);
+    pub const RED: (u8, u8, u8) = (135, 0, 0);
+    pub const ORANGE: (u8, u8, u8) = (215, 95, 0);
+    pub const BLACK: (u8, u8, u8) = (0, 0, 0);
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct Palette {
+    pub source: PaletteSource,
+    pub theme: Theme,
+    pub fg: (u8, u8, u8),
+    pub bg: (u8, u8, u8),
+    pub black: (u8, u8, u8),
+    pub red: (u8, u8, u8),
+    pub green: (u8, u8, u8),
+    pub yellow: (u8, u8, u8),
+    pub blue: (u8, u8, u8),
+    pub magenta: (u8, u8, u8),
+    pub cyan: (u8, u8, u8),
+    pub white: (u8, u8, u8),
+    pub orange: (u8, u8, u8),
+}
+
+impl Default for Palette {
```

Comment:
Can this be done on the Zellij side of things? It's fine to just `#[derive(Default)]` in `zellij-tile` because plugins are guaranteed to receive a `ModeUpdate` when they are loaded. I'd rather this default palette be made in Zellij as opposed to here if possible :)